src/Microsoft.AspNetCore.Mvc.TagHelpers/ScriptTagHelper.cs

```diff
@@ -289,11 +286,7 @@ private void BuildFallbackBlock(TagHelperAttributeList attributes, DefaultTagHel
                        .AppendHtml("||document.write(\"");
 
                 // May have no "src" attribute in the dictionary e.g. if Src and SrcInclude were not bound.
-                if (!attributes.ContainsName(SrcAttributeName))
-                {
-                    // Need this entry to place each fallback source.
-                    attributes.Add(new TagHelperAttribute(SrcAttributeName, value: null));
-                }
+                var addSrc = !attributes.ContainsName(SrcAttributeName);
```

Comment:
+1 Just need a `bool` variable above the `for (var i = 0; i < attributes.Count; i++)` and set to `true` if a `src` attribute is seen.